loader/include.go

```diff
@@ -104,17 +104,27 @@ func ApplyInclude(ctx context.Context, configDetails types.ConfigDetails, model
 			if s, err := os.Stat(f); err == nil && !s.IsDir() {
 				r.EnvFile = types.StringList{f}
 			}
+		} else {
+			envFile := []string{}
+			for _, f := range r.EnvFile {
+				f := filepath.Join(configDetails.WorkingDir, f)
+				if s, err := os.Stat(f); err == nil && !s.IsDir() {
+					envFile = append(envFile, f)
+				}
+			}
+			r.EnvFile = envFile
 		}
 
 		envFromFile, err := dotenv.GetEnvFromFile(configDetails.Environment, r.EnvFile)
 		if err != nil {
 			return err
 		}
+		configDetails.Environment.Merge(envFromFile)
```

Comment:
Why this change ?
Need to clone before merge, otherwise env_file selected to parse the included model will still be present in `configDetails` and will have impacts on next include being processed

Reply:
It took me some time to understand what is the error. Currently, the only environment variable that are resolved DURING the parse is the Interpolation (`$VAR`) and this case `environment: - VAR` will only be solved AFTER the load in https://github.com/compose-spec/compose-go/blob/4c12ed69741da3fc0acecd36bf6504687a95517e/loader/loader.go#L528 
That's why changing the `Merge` to also be applied to the higher level `configDetails` works. However, this is not correct, the environment variables of `custom.env` (from the test and the issue example) should be only within the context of the `include`. I propose to apply a `resolveEnvironment` at the end of the `loadYamlModel` to have this case into consideration, without changing the overall `project.Environment`